core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -215,7 +216,11 @@ public long getDataTimestamp() {
 
     @Restricted(NoExternalUse.class)
     public @NonNull FormValidation updateDirectlyNow(boolean signatureCheck) throws IOException {
-        return updateData(DownloadService.loadJSON(new URL(getUrl() + "?id=" + URLEncoder.encode(getId(), StandardCharsets.UTF_8) + "&version=" + URLEncoder.encode(Jenkins.VERSION, StandardCharsets.UTF_8))), signatureCheck);
+        try {
+            return updateData(DownloadService.loadJSON(new URI(getUrl() + "?id=" + URLEncoder.encode(getId(), StandardCharsets.UTF_8) + "&version=" + URLEncoder.encode(Jenkins.VERSION, StandardCharsets.UTF_8)).toURL()), signatureCheck);
+        } catch (URISyntaxException e) {
+            throw new MalformedURLException(e.getMessage());
```

Comment:
please include the cause of the exception like you have elsewhere.

Reply:
modified to `throw new MalformedURLException("Incorrect URL data returned");`